src/wallet/init.cpp

```diff
@@ -63,7 +63,7 @@ bool WalletParameterInteraction()
         return true;
     }
 
-    gArgs.SoftSetArg("-wallet", DEFAULT_WALLET_DAT);
+    gArgs.SoftSetArg("-wallet", "");
```

Comment:
This means that `m_name` is `""` if no `-wallet` name is specified on the command line or in the .conf file, which makes the `getwalletinfo` and `listwallets` output confusing:

```
bitcoin-cli getwalletinfo
{
  "walletname": "",
  "walletversion": 159900,
  "balance": 0.00000000,
  "unconfirmed_balance": 0.00000000,
  "immature_balance": 0.00000000,
  "txcount": 0,
  "keypoololdest": 1513800503,
  "keypoolsize": 1000,
  "keypoolsize_hd_internal": 1000,
  "paytxfee": 0.00000000,
  "hdmasterkeyid": "4a4cbfcac33849b565a2eb5deecbc96c5c76c06a"
}

→ bitcoin-cli listwallets
[
  ""
]
```

Is it possible to change this so `m_name` is filled to `wallet.dat` when no `wallet` parameter is passed?